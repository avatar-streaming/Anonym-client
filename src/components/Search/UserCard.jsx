import React from "react";
import useCheckFollow from "../../hooks/useCheckFollow";
import useFollow from "../../hooks/useFollow";
import useUnfollow from "../../hooks/useUnfollow";

function UserCard({ _id, thumnail, userName, followers }) {
  const isFollow = useCheckFollow(_id);
  const followUser = useFollow();
  const unfollowUser = useUnfollow();

  return (
    <div className="user-card">
      <div className="user-card__left">
        <img
          className="user-card__thumnail user-thumnail"
          src={thumnail}
          alt="user thumnail"
        />
      </div>
      <div className="user-card__right">
        <div className="user-card__description">
          <div className="user-card__user-name">
            {userName}
          </div>
          <div className="follower-number">
            {followers.length} followers
          </div>
        </div>
        <div className="follow-button-box">
          {
            isFollow ? (
              <button
                className="follow-button"
                onClick={() => {
                  unfollowUser(_id);
                }}
              >
                unfollow
              </button>
            ) : (
              <button
                className="follow-button"
                onClick={() => {
                  followUser(_id);
                }}
              >
                follow
              </button>
            )
          }
        </div>
      </div>
    </div>
  );
}

export default UserCard;
