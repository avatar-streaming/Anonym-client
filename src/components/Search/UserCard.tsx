import useCheckFollow from "../../hooks/useCheckFollow";
import useFollow from "../../hooks/useFollow";
import useUnfollow from "../../hooks/useUnfollow";

<<<<<<< HEAD:src/components/Search/UserCard.tsx
interface UserCardProps {
  _id: string;
  thumnail: string;
  userName: string;
  followers: string[];
}

function UserCard({ _id, thumnail, userName, followers }: UserCardProps) {
=======
function UserCard({ _id, thumnail, userName, followers }) {
>>>>>>> c05b8641dad7cfa992c68cc1b3c69d422f4bf8ce:src/components/Search/UserCard.jsx
  const isFollow = useCheckFollow(_id);
  const followUser = useFollow();
  const unfollowUser = useUnfollow();

  return (
    <div className="user-card">
      <div className="user-card__left">
        <img
          className="user-card__thumnail user-thumnail"
          src={thumnail}
          alt="user thumnail"
        />
      </div>
      <div className="user-card__right">
        <div className="user-card__description">
          <div className="user-card__user-name">
            {userName}
          </div>
          <div className="follower-number">
            {followers.length} followers
          </div>
        </div>
        <div className="follow-button-box">
          {
            isFollow ? (
              <button
                className="follow-button"
                onClick={() => {
                  unfollowUser(_id);
                }}
              >
                unfollow
              </button>
            ) : (
              <button
                className="follow-button"
                onClick={() => {
                  followUser(_id);
                }}
              >
                follow
              </button>
            )
          }
        </div>
      </div>
    </div>
  );
}

export default UserCard;
