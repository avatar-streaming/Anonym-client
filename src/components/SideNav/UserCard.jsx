import React from "react";
import PropTypes from "prop-types";

function UserCard({ userInfo }) {
  return (
    <div className="following-card">
      <div>
        <img src={userInfo.thumnail} className="user-thumnail" alt="user thumnail" />
      </div>
      <div>
        <span className="user-name">{userInfo.userName}</span>
        <span className="streaming-state">Offline</span>
      </div>
    </div>
  );
}

export default UserCard;

UserCard.propTypes = {
  userInfo: PropTypes.shape({
    thumnail: PropTypes.string,
    userName: PropTypes.string,
  }),
};
