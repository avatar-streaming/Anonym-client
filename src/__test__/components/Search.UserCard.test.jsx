import { Provider } from "react-redux";
import thunk from "redux-thunk";
import UserCard from "../../components/Search/UserCard";
import renderer from "react-test-renderer";
import configureMockStore from "redux-mock-store";

describe("<UserCard />", () => {
  const mockStore = configureMockStore([thunk]);
  let store = null;
  let component = null;
  let context = null;

  beforeEach(() => {
    store = mockStore({
      user: {
        userInfo: {
          userName: "mock name 1",
          thumnail: "mock thumnail 1",
          followers: ["mock user2, mock user3"],
          followings: [{
            "_id": 1,
            "_id": 2,
          }],
        },
      },
    });
    store.dispatch = jest.fn();

    context = {
      "_id": 1,
      thumnail: "mock thumnail",
      userName: "mock userName",
      followers: [
        "mock user1",
        "mock user2",
      ],
    };

    component = renderer.create(
      <Provider store={store}>
        <UserCard {...context} />
      </Provider>
    );
  });

  it("should render with given state from redux store and context", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
