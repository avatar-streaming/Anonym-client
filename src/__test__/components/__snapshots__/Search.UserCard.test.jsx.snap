// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserCard /> should render with given state from redux store and context 1`] = `
<div
  className="user-card"
>
  <div
    className="user-card__left"
  >
    <img
      alt="user thumnail"
      className="user-card__thumnail user-thumnail"
      src="mock thumnail"
    />
  </div>
  <div
    className="user-card__right"
  >
    <div
      className="user-card__description"
    >
      <div
        className="user-card__user-name"
      >
        mock userName
      </div>
      <div
        className="follower-number"
      >
        2
         followers
      </div>
    </div>
    <div
      className="follow-button-box"
    >
      <button
        className="follow-button"
        onClick={[Function]}
      >
        follow
      </button>
    </div>
  </div>
</div>
`;
